### PR TITLE
Anisotropy151221

### PR DESCRIPTION
**Pull request by christianegross**

### CODE REVIEW

**Comment by urbach** [comment]
@christianegross which of these pull requests do you want me to merge in?

File: smearape.hh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    for(size_t x1 = 0; x1 < U.getLx(); x1++) {
+      for(size_t x2 = 0; x2 < U.getLy(); x2++) {
+        for(size_t x3 = 0; x3 < U.getLz(); x3++) {
+          std::vector<size_t> x = {x0, x1, x2, x3};  
+          for(size_t mu = startmu; mu < U.getndims(); mu++) {
+            accum K;
+            //~ get_staples(K, Uold, x, mu, xi, anisotropic, spacial);
+            get_staples(K, Uold, x, mu, 1.0, false, spacial);
+            K = K*(1-alpha)/2.0;
+            Group Uprime(Uold(x, mu)*alpha+K);
```
Comment (urbach) [inline smearape.hh:43]: why is this done in two steps here?

File: hmc-u1.cc [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #include"gauge_energy.hh"
 #include"energy_density.hh"
 #include"random_gauge_trafo.hh"
-#include"sweep.hh"
+//#include"sweep.hh"
```
Comment (urbach) [inline hmc-u1.cc:7]: remove if not needed (why commented out?)

File: hmc.cc [hunk 3]
```diff
@@ -2,7 +2,7 @@
 #include"gaugeconfig.hh"
 #include"gauge_energy.hh"
 #include"random_gauge_trafo.hh"
-#include"sweep.hh"
+//#include"sweep.hh"
```
Comment (urbach) [inline hmc.cc:5]: same again.

File: kramers.cc [hunk 4]
```diff
@@ -2,7 +2,7 @@
 #include"gaugeconfig.hh"
 #include"gauge_energy.hh"
 #include"random_gauge_trafo.hh"
-#include"sweep.hh"
+//#include"sweep.hh"
```
Comment (urbach) [inline kramers.cc:5]: and again

File: measure-u1.cc [hunk 5]
```diff
@@ -3,7 +3,7 @@
 #include"gaugeconfig.hh"
 #include"gauge_energy.hh"
 #include"random_gauge_trafo.hh"
-#include"sweep.hh"
+//~ #include"sweep.hh"
```
Comment (urbach) [inline measure-u1.cc:6]: and again

File: measure.cc [hunk 6]
```diff
@@ -2,7 +2,7 @@
 #include"gaugeconfig.hh"
 #include"gauge_energy.hh"
 #include"random_gauge_trafo.hh"
-#include"sweep.hh"
+//#include"sweep.hh"
```
Comment (urbach) [inline measure.cc:5]: and once more

File: potentialu1.cc [hunk 7]
```diff
@@ -0,0 +1,166 @@
+#include"su2.hh"
```
Comment (urbach) [inline potentialu1.cc:1]: does this require a new executable? Cannot be included in measure?

File: sweep.hh [hunk 8]
```diff
@@ -5,42 +5,98 @@
...
+  #pragma omp parallel
+  {
+    int thread_num = omp_get_thread_num();
+  #else
+  int thread_num=0;  
+  #endif  
+  size_t temp=0;
+  #pragma omp for
+  for(size_t x0 = 0; x0 < U.getLt(); x0+=2) {
+//OpenMP does not allow loop declaration as it was done for the other dimensions, still have to figure out why
```
Comment (urbach) [inline sweep.hh:35]: what is it that cannot be done here?

File: tryreduce.cc [hunk 9]
```diff
@@ -0,0 +1,47 @@
+#ifdef _USE_OMP_
+#  include<omp.h>
+#endif
+#include<iostream>
+
```
Comment (urbach) [inline tryreduce.cc:5]: what is this function supposed to do?

File: u1.hh [hunk 10]
```diff
@@ -14,10 +14,12 @@ public:
   explicit _u1() : a(0) {}
   explicit _u1(double _a) : a(_a) {}
   _u1(const _u1& U) : a(U.a) {}
+  _u1(Complex _a) : a(std::arg(_a)) {}
 
   friend Complex operator+(const _u1 &U1, const _u1 &U2);
   friend Complex operator-(const _u1 &U1, const _u1 &U2);
   friend _u1 operator*(const _u1 &U1, const _u1 &U2);
+  friend _u1 operator+(const _u1 &U1, const Complex c);
```
Comment (urbach) [inline u1.hh:22]: this is in general no more in U(1), is it?

**Review by urbach — CHANGES_REQUESTED** [review]
should merge this soon, up to the smallish comments...

File: tryreduce.cc (hunk 9 above)
Comment (christianegross) [reply tryreduce.cc:5]: valgrind found some memory leaks, and I wanted to see if they were coming from my code or from somewhere else. They also appear in this minimum example.

**Review by urbach — APPROVED** [review]

**Pull request merged**